docker/docker_run.go

```
@@ -0,0 +1,302 @@
+	"golang.org/x/net/context"
+)
+
+type DockerRunStep struct {
+	*core.BaseStep
+	options               *core.PipelineOptions
+	dockerOptions         *Options
+	data                  map[string]string
+	env                   []string
+	logger                *util.LogEntry
+	Cmd                   []string
+	EntryPoint            []string
+	WorkingDir            string
+	PortBindings          map[docker.Port][]docker.PortBinding
+	ExposedPorts          map[docker.Port]struct{}
+	User                  string
+	ContainerName         string
+	OriginalContainerName string
+	Image                 string
+	ContainerID           string
+	Repository            string
+	Auth                  dockerauth.CheckAccessOptions `yaml:",inline"`
+}
+
+type BoxDockerRun struct {
+	*DockerBox
+}
+
+// NewDockerRunStep is a special step for doing docker runs
+func NewDockerRunStep(stepConfig *core.StepConfig, options *core.PipelineOptions, dockerOptions *Options) (*DockerRunStep, error) {
+	name := "docker-run"
+	displayName := "docker run"
+	originalContainerName := name
+	if stepConfig.Name != "" {
+		originalContainerName = stepConfig.Name
+	}
+
+	// Add a random number to the name to prevent collisions on disk
+	stepSafeID := fmt.Sprintf("%s-%s", name, uuid.NewRandom().String())
+
+	baseStep := core.NewBaseStep(core.BaseStepOptions{
+		DisplayName: displayName,
+		Env:         &util.Environment{},
+		ID:          name,
+		Name:        name,
+		Owner:       "wercker",
+		SafeID:      stepSafeID,
+		Version:     util.Version(),
+	})
+
+	return &DockerRunStep{
+		BaseStep:              baseStep,
+		data:                  stepConfig.Data,
+		logger:                util.RootLogger().WithField("Logger", "DockerRunStep"),
+		options:               options,
+		dockerOptions:         dockerOptions,
+		OriginalContainerName: originalContainerName,
+	}, nil
+}
+
+// InitEnv parses our data into our config
+func (s *DockerRunStep) InitEnv(env *util.Environment) {
```

If you will merge with master now - you will need to return an error from here..